src/main/java/io/codelex/flightplanner/service/FlightPlannerDatabaseService.java

```diff
@@ -0,0 +1,134 @@
+import io.codelex.flightplanner.response.PageResult;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+public class FlightPlannerDatabaseService implements FlightPlannerService{
+    private final FlightDatabaseRepository flightDatabaseRepository;
+    private final AirportDatabaseRepository airportDatabaseRepository;
+    private final Airport from = new Airport();
+    private final Airport to = new Airport();
+
+    public FlightPlannerDatabaseService(FlightDatabaseRepository flightDatabaseRepository, AirportDatabaseRepository airportDatabaseRepository) {
+        this.flightDatabaseRepository = flightDatabaseRepository;
+        this.airportDatabaseRepository = airportDatabaseRepository;
+    }
+
+    @Override
+    public FlightResponse fetchFlight(long id) {
+        Optional<Flight> flight = flightDatabaseRepository.findById(id);
+        return new FlightResponse(flight.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)));
+    }
+
+    @Override
+    public void clear() {
+        flightDatabaseRepository.deleteAll();
+        airportDatabaseRepository.deleteAll();
+    }
+
+    @Override
+    public synchronized Flight addFlight(AddFlightRequest flightRequest) {
+        Flight flight = isValidAddFlightRequest(flightRequest);
+        this.flightDatabaseRepository.save(flight);
+        return flight;
+    }
+
+    @Override
+    public Flight isValidAddFlightRequest(AddFlightRequest flightRequest) {
+        if (isValidAirport(flightRequest)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+        if (!isValidTime(flightRequest)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+        if (isValidFlightRequest(flightRequest)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+        return createFlight(flightRequest);
+    }
+
+    @Override
+    public boolean isValidAirport(AddFlightRequest flightRequest) {
+        String from = flightRequest.getFrom().getAirport().toLowerCase().trim();
+        String to = flightRequest.getTo().getAirport().toLowerCase().trim();
+
+        return from.equals(to);
+    }
+
+    @Override
+    public boolean isValidTime(AddFlightRequest flightRequest) {
+        TimeDTO flightTimeDTO = new TimeDTO(flightRequest.getDepartureTime(), flightRequest.getArrivalTime());
+        return flightTimeDTO.isBefore();
+    }
+
+    @Override
+    public boolean isValidFlightRequest(AddFlightRequest flightRequest) {
+        Flight flight = createFlight(flightRequest);
+        return flightDatabaseRepository.findAll().contains(flight);
+    }
+
+    @Override
+    public void deleteFlight(long id) {
+        flightDatabaseRepository.deleteById(id);
+    }
+
+    public Flight createFlight(AddFlightRequest addFlightRequest) {
+        addAirports(addFlightRequest);
```

Comment:
Šis ir kaut kas dīvains, sanāk Tev ienāk request ar visām lidostām un visu un tas ko izdari ir sauc AddAirports, kas izņem no request, kurā jau bija divi airport, datus, saliek tos klases līmeņa mainīgajos un pēc tam pašā flight ieliek iekšā tos klases līmeņa mainīgos. 🤔  Tā vietā vienkārši no request paņem from un to un ieliec iekš Flight. Ko vienīgais iespējams vajag ir pārbaudīt pirms tam vai tāds airport pēc koda jau neeksistē DB, un ja ne - tad jāsaglabā iepriekš atsevišķi DB, ja eksistē, tad jāpaņem no DB.

Reply:
Cik zinu, save metode pati to dara automātiski, un vēlreiz pārbaude nav jāveic. Vēl  pie tam, man airport (kods) ir Primary key iekš DB tabulas, kas nodrošina to unikalitāti. Protam,s ir viena problēma - ja iepāso vienu lowercase, vienu uppercase, tad gan jau vajag to pārbaudīt, vai tomēr ne?